File: 백준/Silver/1522. 문자열 교환/문자열 교환.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String line = br.readLine();
		int min = Integer.MAX_VALUE;
		
		int aCnt = 0;
		for (int i = 0; i < line.length(); i++) {
			if (line.charAt(i) == 'a')
				aCnt++;
		}
		
		for (int i = 0; i < line.length(); i++) {
			int bCnt = 0;
			for (int j = i; j < i + aCnt; j++) {
				if (line.charAt(j % line.length()) =='b')
					bCnt++;
			}
			min = Math.min(min, bCnt);
		}
		System.out.println(min);
	}

}

File: 백준/Silver/1522. 문자열 교환/문자열 교환.py
import sys
input = sys.stdin.readline


arr = list(input().rstrip())
size = len(arr)

a_cnt = 0
for i in range(size):
    if arr[i] == 'a':
        a_cnt += 1

ans = 1000000
for i in range(size):
    b_cnt = 0
    for j in range(i, i + a_cnt):
        if arr[j % size] == 'b':
            b_cnt += 1
    ans = min(ans, b_cnt)

print(ans)
